nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/internal/metadata/ArtifactsXmlAbsoluteUrlRemover.java

```diff
@@ -77,38 +79,41 @@
 
   private static final String ARTIFACTS_XML = "artifacts.xml";
 
-  public TempBlob removeMirrorUrlFromArtifactsXml(final TempBlob artifact,
-                                                  final Repository repository,
-                                                  final String extension)
+  private static final String REPOSITORY = "repository";
+
+  public TempBlob removeMirrorUrlFromArtifactsXml(
+      final TempBlob artifact,
+      final Repository repository,
+      final String extension)
   {
     try {
       Path tempFile = createTempFile("p2-artifacts-" + randomUUID().toString(), "xml");
```

Comment:
Use something like 
  private Path getTemporaryFile(String prefix, String suffix) {
    Path tempFile;
    try {
      tempFile = createTempFile(prefix, suffix);
    }
    catch (IOException e) {
      log.error("Unable to create temporary file");
      throw new RuntimeException(e);
    }
    return tempFile;
  }